oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoder.java

```diff
@@ -274,7 +284,7 @@ public static final class JwkSetUriJwtDecoderBuilder {
 		private static final JOSEObjectTypeVerifier<SecurityContext> NO_TYPE_VERIFIER = (header, context) -> {
 		};
 
-		private final Function<RestOperations, String> jwkSetUri;
+		private Function<RestOperations, String> jwkSetUri;
```

Comment:
Since providing a `JWKSource` supercedes a `RestOperations` and a `Cache`, please create `JwkSourceJwtDecoderBuilder` instead of repurposing `JwkSetUriJwtDecoderBuilder`. In that way, people won't provide a `JWKSource` and then also try and provide a `RestOperations` or `Cache`.

Reply:
Note that `JwkSourceJwtDecoderBuilder` will not need the `validateTypes` flag since that was added as part of migrating from 6.x to 7.0. Since this class will be added in 7.0, it doesn't need the same transition flag.